cpp/src/parquet/column_writer.cc

```diff
@@ -1346,7 +1346,12 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
 
   EncodedStatistics GetChunkStatistics() override {
     EncodedStatistics result;
-    if (chunk_statistics_) result = chunk_statistics_->Encode();
+    if (chunk_statistics_) {
+      if (has_dictionary_ && !fallback_) {
+        chunk_statistics_->SetDistinctCount(current_dict_encoder_->num_entries());
```

Comment:
@pitrou I think this is the proper way to set distinct count. Another method is

```c++
if (chunk_statistics_) { 
  result = chunk_statistics_->Encode();
  if (has_dictionary_ && !fallback_) {
    result.set_distinct_count(...);
  }
}
```

Do you think this is ok?

Reply:
I _think_ it's ok. But I don't think it's very important to write the distinct count, if we're only able to write it for dictionary-encoded chunks.

Reply:
Ok so let me just remove it. I can use `distinct_count` to inspect dictionary, but I guess it's really useless